internal/controller/providers/psmdb/applier.go

```diff
@@ -366,10 +366,17 @@ func (p *applier) applyPMMCfg(monitoring *everestv1alpha1.MonitoringConfig) erro
 		return err
 	}
 
+	// We avoid setting the controller reference for the Secret here.
+	// The PSMDB operator handles Secret cleanup automatically as part of the `delete-psmdb-pvc` finalizer process.
+	// If we were to set the controller reference, the Secret would be deleted before the PSMDB cluster, leading to
+	// the PSMDB operator recreating the Secret and restarting the DB pods to sync updated user information,
+	// which would cause unnecessary restarts.
+	setControllerRef := false
```

Comment:
This variable is not actually required, we can simply pass `false` to the function parameter, but I added it for the sake of readability only - there are many parameters in the function call and its not immediately obvious that the `false` refers to the controller reference setting..